lib/parse-styles.js

```diff
@@ -165,6 +165,8 @@ async function loadImportContent(
       state.importedFiles[filename][media] = {}
     }
     state.importedFiles[filename][media][layer] = true
+  } else if (from.includes(filename)) {
+    return
   }
```

Comment:
```suggestion
  }

  if (from.includes(filename)) {
    return
  }
```

Even with `skipDuplicates: true` there can be import cycles when they have alternating conditions.

```css
/* a.css */
@import url("b.css") screen;
```

```css
/* b.css */
@import url("a.css") all;
```

The conditions will always be joined with `and` creating an ever longer but always unique condition : `screen and all and screen and all and ...`

Reply:
needs a test case